airbyte-webapp/src/views/Connector/ConnectorCard/components/TestCard.tsx

```diff
@@ -0,0 +1,119 @@
+import { Card } from "components/ui/Card";
+import { FlexContainer, FlexItem } from "components/ui/Flex";
+import { ProgressBar } from "components/ui/ProgressBar";
+import { Text } from "components/ui/Text";
+
+import { SynchronousJobRead } from "core/request/AirbyteClient";
+
+import { TestingConnectionError } from "../../ConnectorForm/components/TestingConnectionError";
+import styles from "./TestCard.module.scss";
+import TestingConnectionSuccess from "./TestingConnectionSuccess";
+
+interface IProps {
+  formType: "source" | "destination";
+  isValid: boolean;
+  dirty: boolean;
+  onRetestClick: () => void;
+  onCancelTesting: () => void;
+  isTestConnectionInProgress?: boolean;
+  successMessage?: React.ReactNode;
+  errorMessage?: React.ReactNode;
+  job?: SynchronousJobRead;
+  isEditMode?: boolean;
+  connectionTestSuccess: boolean;
+}
+
+const PROGRESS_BAR_TIME = 60 * 2;
+
+export const TestCard: React.FC<IProps> = ({
+  isTestConnectionInProgress,
+  isValid,
+  formType,
+  onRetestClick,
+  connectionTestSuccess,
+  errorMessage,
+  onCancelTesting,
+  job,
+  dirty,
+  isEditMode,
+}) => {
+  const [logsVisible, setLogsVisible] = useState(false);
+
+  const renderStatusMessage = () => {
+    if (errorMessage) {
+      return (
+        <FlexContainer direction="column">
+          <TestingConnectionError errorMessage={errorMessage} />
+          {job && (
+            <div>
+              <Button
+                variant="clear"
+                type="button"
+                icon={<FontAwesomeIcon icon={logsVisible ? faChevronDown : faChevronRight} />}
+                onClick={() => {
+                  setLogsVisible(!logsVisible);
+                }}
+              >
+                <FormattedMessage id="connector.testLogs" />
+              </Button>
+              {logsVisible && <JobLogs job={job} jobIsFailed />}
+            </div>
+          )}
+        </FlexContainer>
+      );
+    }
+    if (connectionTestSuccess) {
+      return <TestingConnectionSuccess />;
+    }
+    return null;
+  };
+
+  return (
+    <Card className={styles.cardTest}>
+      <FlexContainer direction="column">
+        <FlexContainer alignItems="center">
+          <FlexItem grow>
+            <Text size="lg">
+              <FormattedMessage id={`form.${formType}RetestTitle`} />
+            </Text>
+          </FlexItem>
+          {isTestConnectionInProgress ? (
+            <Button
+              className={styles.button}
+              icon={<FontAwesomeIcon icon={faClose} />}
+              variant="secondary"
+              type="button"
+              onClick={() => onCancelTesting?.()}
+            >
+              <FormattedMessage id="form.cancel" />
+            </Button>
+          ) : (
+            <Button
+              type="button"
+              onClick={onRetestClick}
+              variant="secondary"
+              icon={<FontAwesomeIcon icon={faRefresh} />}
+              disabled={!isValid || (!isEditMode && !dirty)}
```

Comment:
```suggestion
              disabled={!isValid || (isEditMode && !dirty)}
```
I think this was a mistake. The current logic means that if you set up a new connector and don't change any of the values (e.g set up an E2E Testing destination and don't change any of the defaults), then it won't let you test or submit.

Reply:
Good catch, actually it's even simpler - it should always be possible to re-test as long as the form is valid right now.